src/Microsoft.AspNet.Routing/Internal/DecisionTree/DecisionCriterion.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Routing.Internal.DecisionTree
+{
+    public class DecisionCriterion<TItem>
```

Comment:
Wait, stop, we don't want to make a copy of this code. This is also used in action selection.